docs/src/components/home/MaterialDesignComponents.tsx

```diff
@@ -432,7 +432,7 @@ export function buildTheme(theme: Theme): ThemeOptions {
       },
       MuiTooltip: theme.components?.MuiTooltip,
       MuiPaper: theme.components?.MuiPaper,
-      MuiTableCell: deepmerge(theme.components?.MuiTableCell, {
+      MuiTableCell: deepmerge(theme.components?.MuiTableCell || {}, {
```

Comment:
Hmm, is this really required? If yes, this looks like a breaking change

Reply:
Yes, I'm going to roll back here, I prepared a new commit, but I have a doubt, currently `const output = options.clone ? { ...target } : target;`, if I pass in an array [0, 1] This returns {0:1, 1:1}, I'm not sure this is the intended result?

Reply:
Actually as far as I can see from the implementation, we only support merging of objects. We are spreading the objects before checking if they are both plain objects. In my opinion we should be explicit that we support only merging of objects as the implementation clearly does not support anything else. Good catch btw.